src/scripts/sections/product.js

```diff
@@ -6,217 +6,202 @@
-    $thumbnail.attr('aria-current', true);
+    this.renderFeaturedImage(variant.featured_image.id);
+    this.renderActiveThumbnail(variant.featured_image.id);
   },
 
-  switchImage(imageId) {
-    const $newImage = $(
-      `${selectors.productImageWrapper}[data-image-id='${imageId}']`,
-      this.$container,
+  renderPrice(variant) {
+    const priceElement = this.container.querySelector(selectors.productPrice);
+    const priceWrapperElement = this.container.querySelector(
+      selectors.priceWrapper,
     );
-    const $otherImages = $(
-      `${selectors.productImageWrapper}:not([data-image-id='${imageId}'])`,
-      this.$container,
-    );
-    $newImage.removeClass(cssClasses.hide);
-    $otherImages.addClass(cssClasses.hide);
-  },
 
-  /**
-   * Updates the DOM state of the add to cart button
-   *
-   * @param {boolean} enabled - Decides whether cart is enabled or disabled
-   * @param {string} text - Updates the text notification content of the cart
-   */
-  updateAddToCartState(evt) {
-    const variant = evt.variant;
+    priceWrapperElement.classList.toggle(classes.hide, !variant);
 
     if (variant) {
-      $(selectors.priceWrapper, this.$container).removeClass(cssClasses.hide);
-    } else {
-      $(selectors.addToCart, this.$container).prop('disabled', true);
-      $(selectors.addToCartText, this.$container).html(
-        theme.strings.unavailable,
-      );
-      $(selectors.priceWrapper, this.$container).addClass(cssClasses.hide);
+      priceElement.innerText = formatMoney(variant.price, theme.moneyFormat);
+    }
+  },
+
+  renderComparePrice(variant) {
+    if (!variant) {
       return;
     }
 
-    if (variant.available) {
-      $(selectors.addToCart, this.$container).prop('disabled', false);
-      $(selectors.addToCartText, this.$container).html(theme.strings.addToCart);
+    const comparePriceElement = this.container.querySelector(
+      selectors.comparePrice,
+    );
+    const compareTextElement = this.container.querySelector(
+      selectors.comparePriceText,
+    );
+
+    if (variant.compare_at_price > variant.price) {
+      comparePriceElement.innerText = formatMoney(
+        variant.compare_at_price,
+        theme.moneyFormat,
+      );
+      compareTextElement.classList.remove(classes.hide);
+      comparePriceElement.classList.remove(classes.hide);
     } else {
-      $(selectors.addToCart, this.$container).prop('disabled', true);
-      $(selectors.addToCartText, this.$container).html(theme.strings.soldOut);
+      comparePriceElement.innerText = '';
+      compareTextElement.classList.add(classes.hide);
+      comparePriceElement.classList.add(classes.hide);
     }
   },
 
-  updateImages(evt) {
-    const variant = evt.variant;
-    const imageId = variant.featured_image.id;
+  renderActiveThumbnail(id) {
+    const thumbnailToSetActive = this.container.querySelector(
+      selectors.thumbnailById(id),
+    );
+    const thumbnailToSetDeactive = this.container.querySelector(
```

Comment:
Not a huge fan of the name `thumbnailToSetDeactive`; maybe `activeThumbnail` and `inactiveThumbnail` instead?